hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/net/NetUtils.java

```diff
@@ -39,13 +39,16 @@
 import java.nio.channels.SocketChannel;
 import java.nio.channels.UnresolvedAddressException;
 import java.util.Map.Entry;
+import java.util.concurrent.TimeUnit;
 import java.util.regex.Pattern;
 import java.util.*;
 import java.util.concurrent.ConcurrentHashMap;
 
 import javax.net.SocketFactory;
 
 import org.apache.hadoop.security.AccessControlException;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
```

Comment:
Does Hadoop have import order checking and specification?

Reply:
I don't believe checkstyle checks for it or warns about it, however it is good practice to keep them ordered. I sorted them in Intellij and there are quite a few out of order, so I think it may cause more problems that it fixes if we sort them all.

However in checking with Intellij, it moved com.google.common.cache.Cache -> org.apache.hadoop.thirdparty.com.google.common.cache.Cache. This was part of HADOOP-17288 which is on branch-3.3.

I will push a new commit with the new thirdparty change, but I think we should just leave the order as it is. What do you think?

Reply:
thanks for your address. LGTM.